internal/resources/kustomization/resource_kustomization_test.go

```diff
@@ -0,0 +1,227 @@
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/proxy"
+	kustomizationclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kustomization/cluster"
+	kustomizationclustergroupmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kustomization/clustergroup"
+	commonscope "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common/scope"
+	gitrepositoryhelper "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/gitrepository"
+	kustomizationscope "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kustomization/scope"
+	testhelper "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/testing"
+)
+
+// nolint: gosec
+const (
+	kustomizationResource      = ResourceName
+	kustomizationResourceVar   = "test_kustomizationy"
+	kustomizationDataSourceVar = "test_data_source_kustomizationy"
+	kustomizationNamePrefix    = "tf-kustomization-test"
+
+	gitRepositoryResource    = gitrepositoryhelper.ResourceName
+	gitRepositoryResourceVar = "test_git_repository"
+	gitRepositoryNamePrefix  = "tf-gr-test"
+)
+
+type testAcceptanceConfig struct {
+	Provider                  *schema.Provider
+	KustomizationResource     string
+	KustomizationResourceVar  string
+	KustomizationResourceName string
+	KustomizationName         string
+	ScopeHelperResources      *commonscope.ScopeHelperResources
+	GitRepositoryResource     string
+	GitRepositoryResourceVar  string
+	GitRepositoryName         string
+}
+
+func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
+	return &testAcceptanceConfig{
+		Provider:                  initTestProvider(t),
+		KustomizationResource:     kustomizationResource,
+		KustomizationResourceVar:  kustomizationResourceVar,
+		KustomizationResourceName: fmt.Sprintf("%s.%s", kustomizationResource, kustomizationResourceVar),
+		KustomizationName:         acctest.RandomWithPrefix(kustomizationNamePrefix),
+		ScopeHelperResources:      commonscope.NewScopeHelperResources(),
+		GitRepositoryResource:     gitRepositoryResource,
+		GitRepositoryResourceVar:  gitRepositoryResourceVar,
+		GitRepositoryName:         acctest.RandomWithPrefix(gitRepositoryNamePrefix),
+	}
+}
+
+// nolint: dupl
+func TestAcceptanceForKustomizationResource(t *testing.T) {
+	testConfig := testGetDefaultAcceptanceConfig(t)
+
+	t.Log("start kustomization resource acceptance tests!")
+
+	// Test case for kustomization resource for GO_GIT git implementation type.
+	resource.Test(t, resource.TestCase{
```

Comment:
Take reference of network policy PR and add update scenario in the tests. 
cc: @ramya-bangera

Reply:
added PTAL